src/lib/merkle_ledger/database.ml

```diff
@@ -700,6 +700,58 @@ module Make (Inputs : Inputs_intf) :
     List.map2_exn dependency_dirs dependency_hashes ~f:(fun dir hash ->
         Direction.map dir ~left:(`Left hash) ~right:(`Right hash) )
 
+  let merkle_path_batch mdb locations =
```

Comment:
@ghost-not-in-the-shell could you check the version I have in #14594

I also wanted to refactor it but did that at a higher layer of PR chain to avoid conflicts

Reply:
But your version is even easier

Reply:
Do you mind if I use your code at PR #14594 and keep this one as it is?

Reply:
In fact version in that PR becomes a bit more general because of wide paths. ANyway, I suggest having a discussion on final version there as here is only an intermediate step and overall this function as here seems to eb correct.